src/app/qgisapp.cpp

```diff
@@ -2828,7 +2831,26 @@ void QgisApp::createToolBars()
   };
   moveFeatureButton->setDefaultAction( defAction );
   connect( moveFeatureButton, &QToolButton::triggered, this, &QgisApp::toolButtonActionTriggered );
-  mDigitizeToolBar->insertWidget( mActionVertexTool, moveFeatureButton );
+  mDigitizeToolBar->insertWidget( mActionDeleteSelected, moveFeatureButton );
+
+  // vertex tool button
+  QToolButton *vertexToolButton = new QToolButton( mDigitizeToolBar );
+  vertexToolButton->setPopupMode( QToolButton::MenuButtonPopup );
+  vertexToolButton->addAction( mActionVertexTool );
+  vertexToolButton->addAction( mActionVertexToolActiveLayer );
+  mDigitizeToolBar->insertWidget( mActionDeleteSelected, vertexToolButton );
+  QAction *defActionVertexTool = mActionVertexTool;
+  switch ( settings.value( QStringLiteral( "UI/defaultVertexTool" ), 0 ).toInt() )
```

Comment:
```c++
settings.enumValue( QStringLiteral( "UI/defaultVertexTool" ), QgsVertexTool::AllLayers );
```
(or ActiveLayer, but not sure what is agreed as default here, would prefer AllLayers)

Reply:
I don't have a strong opinion here, but I think that we should keep the `AllLayers` mode too.